### ISSUE TITLE
 Split - Unexpected error shows up when creating Scan bill split in workspace room

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32376]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.7-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to + > New chat > Room
3. Create a 'Workspace' visibility room
4. Go to the room and invite a user by mentioning them
5. Click > + > Split bill
6. Create a Scan bill split
7. Click on the split preview
8. Manually enter amount and merchant > Split

## Expected Result:
Bill split is created successfully

## Actual Result:
Unexpected error shows up when creating bill split

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/f47b9ae5-ca49-4985-a8f7-dc3dc710a8cd

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by studentofcoding** [comment]
Action performed for steps 1-4 are not shown on the Video, and thus can't be replicated. It'll be great if we can update the video to include it @lanitochka17

**Comment by keisyrzk** [comment]
I have created the room and followed the split steps the same way as on the provided video. I tried multiple times and could not reproduce the described issue.

**Comment by bfitzexpensify** [comment]
Also struggling to reproduce - @lanitochka17, are you able to do what @studentofcoding has suggested here?

>Action performed for steps 1-4 are not shown on the Video, and thus can't be replicated. It'll be great if we can update the video to include it @lanitochka17

**Comment by lanitochka17** [comment]
Steps 1-4 are shown on the Video
 I am able to reproduce an issue on latest build 1.4.9-2
 
 

https://github.com/Expensify/App/assets/78819774/218b7a17-a2fa-4791-9881-e65304977352

**Comment by keisyrzk** [comment]
Hello, still cannot reproduce on staging.

https://github.com/Expensify/App/assets/18525360/3cf699d8-32be-477a-8ba3-1dd06e25276a

**Comment by keisyrzk** [comment]
My ideas what might go wrong:
- my account is quite fresh, maybe yours are older and the backend pushes the failure for some older accounts. Maybe it is connected with data migration or some "account ID" logic?
- maybe you already have some exact action performed along side with the users you invite to split the bill? Maybe you have already done it before and the backend fails in such case? It might also be some remaining in database so the backend fails
- backend may not find one of the users id and fails

I have tried multiple times on different platforms (mac chrome, ios and android chrome) and all worked as intended each time.
What if you try with a newly created accounts (one for splitting the bill and one fot sharing with). I believe this is a backend error. The question is why the backend fails. I believe in some particular case the app sends incomplete data, in wrong format or so or the backend itself fails/ I would suggest investigating backned first there we may more accurate find out where, when and what happens.

**Comment by bfitzexpensify** [comment]
Thanks for the details thoughts @keisyrzk - I'll try with new accounts (both for creating/splitting the bill and a new account to share to) as that's an interesting theory.

**Comment by bfitzexpensify** [comment]
I've tested again with a new account (both the account creating the workspace/room/split, and the invited account to split the bill with) - I can still reproduce this reliably. 

@eVoloshchak are you able to reproduce this?

**Comment by bfitzexpensify** [comment]
Bump @eVoloshchak, can you still reproduce?

**Comment by bfitzexpensify** [comment]
Another friendly bump here @eVoloshchak - can you reproduce with new accounts?

**Comment by eVoloshchak** [comment]
Was able to reproduce this issue using steps from the issue description and a a couple of new accounts

https://github.com/Expensify/App/assets/9059945/657f476e-172c-4d16-9511-e639c5d93809

**Comment by bfitzexpensify** [comment]
OK, so this is still valid. Let's double the bounty since we have no proposals so far.

**Comment by FlorianWueest** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue: 
Error message is displaying when we try to edit a split bill. 

### What is the root cause of that problem?
Error is reproducible in a slow internet setting (throttled connection). When edited, 'genericCreateFailureMessage' is displaying. 

Error seems to be happening here: 

https://github.com/Expensify/App/blob/6365aebb9cdb3dd4a66e77117afd0fb1b1fa8957/src/pages/iou/SplitBillDetailsPage.js#L107-L110

The 2nd failureData index, not with 'unmodified transaction': 

https://github.com/Expensify/App/blob/6365aebb9cdb3dd4a66e77117afd0fb1b1fa8957/src/libs/actions/IOU.js#L1840-L1859

### What changes do you think we should make in order to solve the problem?
Haven't yet figured out to properly prevent the issue. Will look at this again later today if it hasn't been solved.

**Comment by bfitzexpensify** [comment]
How do these proposals look @eVoloshchak?

**Comment by bfitzexpensify** [comment]
Bump here @eVoloshchak

**Comment by eVoloshchak** [comment]
I agree with @bernhardoj's [alternative solution](https://github.com/Expensify/App/issues/32376#issuecomment-1870044530)

> Support splitting bill with an unvalidated user on the BE.

If it works properly for manual requests, it should work for the split bill too.
This needs a back-end fix, @bfitzexpensify, could you make this internal please?

**Comment by bfitzexpensify** [comment]
>Support splitting bill with an unvalidated user on the BE.

Sounds similar to https://github.com/Expensify/App/issues/30140? What do you think @bernhardoj @eVoloshchak?

**Comment by bernhardoj** [comment]
Looks like it's different. If I understand correctly, in https://github.com/Expensify/App/issues/30140, when we create a split bill for the 2nd time, a new DM chat between the participants is created again even though we already have one, thus we get the error "There is a previously existing chat between these users".

**Comment by bfitzexpensify** [comment]
Thanks @bernhardoj - posted this in [Slack](https://expensify.slack.com/archives/C05RECHFBEW/p1704718075781149)

**Comment by bfitzexpensify** [comment]
Waiting for an internal volunteer on this one

**Comment by bfitzexpensify** [comment]
I'm going to be mostly ooo until Jan 29, assigning a second BZ team member to keep an eye on this. 

- we've identified this is a BE issue
- it's been added to the #vip-split issue https://github.com/orgs/Expensify/projects/111
- needs an internal volunteer to take it on

**Comment by joekaufmanexpensify** [comment]
Got it. Sounds good!

**Comment by joekaufmanexpensify** [comment]
Not overdue. Will try and find a volunteer this week

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
[Discussing. ](https://expensify.slack.com/archives/C05RECHFBEW/p1705678027681349)

**Comment by joekaufmanexpensify** [comment]
Notes for anyone internal who picks this up:

- I just retested and the steps/video in OP to reproduce this bug are still valid. 
- Notes from C+ on this needing to be fixed backend are [here. ](https://github.com/Expensify/App/issues/32376#issuecomment-1876016622)

**Comment by joekaufmanexpensify** [comment]
Auth PR is already out on prod. Confirming [here](https://expensify.slack.com/archives/C05RECHFBEW/p1705935836509819?thread_ts=1705678027.681349&cid=C05RECHFBEW) whether anything else is needed for this.
